lib/trino-parquet/src/main/java/io/trino/parquet/writer/ParquetWriter.java

```diff
@@ -216,6 +216,7 @@ public void close()
             flush();
             writeFooter();
         }
+        columnWriters.clear();
```

Comment:
FYI I had attempted similar change in https://github.com/trinodb/trino/commit/822f43e00c2db438291cbcbc4dac5c64cd426766
It resulted in a query failure (caught by benchmark suites outside of trino repo tests) and I had to revert it in https://github.com/trinodb/trino/pull/13263 (see error stack trace in the PR)
cc: @sopel39